src/ui/Base/TopNavigation.tsx

```diff
@@ -0,0 +1,71 @@
+import { IconLeftArrow, IconOptions, IconPost, IconShare } from '@public/icons';
+
+interface Props {
+  variant?: 'primary' | 'secondary';
+  backgroundColor?: 'bg-background' | 'bg-white';
+  backButton?: boolean;
+  onClickBack?: () => void;
+  title?: string;
+  option?: 'none' | 'share' | 'post';
+  onClickOption?: () => void;
+  isOwner?: boolean;
+  onClickOwner?: () => void;
+}
+
+const TopNavigation = ({
+  variant = 'primary',
+  backgroundColor = 'bg-background',
+  backButton = true,
+  onClickBack,
+  title = '',
+  option = 'none',
+  onClickOption,
+  isOwner = false,
+  onClickOwner,
+}: Props) => {
+  return (
+    <div className={`${BASE_CLASSES} ${backgroundColor}`}>
+      <div className="flex">
+        {backButton && (
+          <div className={`${ICON_CLASSES}`}>
+            <IconLeftArrow onClick={onClickBack} />
+          </div>
+        )}
+      </div>
+      <div className={`${TITLE_CLASSES} ${VARIANT_CLASSES[variant]}`}>
+        {title}
+      </div>
+      <div className="flex gap-[1rem]">
+        {option !== 'none' && (
+          <div className={`${ICON_CLASSES}`}>
+            {option === 'share' ? (
+              <IconShare onClick={onClickOption} />
+            ) : (
+              <IconPost onClick={onClickOption} />
+            )}
+          </div>
+        )}
```

Comment:
help;

option에 대한 결과값이 3개가 있는데 조건문이 2번 겹쳐 한눈에 안들어오네요
다른 방법이 있을까요 🤔 
```suggestion
        {option !== 'none' && (
          <div className={`${ICON_CLASSES}`}>
            {option === 'share' && (
              <IconShare onClick={onClickOption} />
            )};
            {option === 'post' && (
              <IconPost onClick={onClickOption} />
            )};
          </div>
        )};
```
??? 🤔

Reply:
코멘트 의도와 조금 다른 의견이지만, 저는 아이콘을 렌더링하는 부분은 각각의 페이지에서 처리를 해주어야 한다고 생각해요.

지금처럼 prop을 통해 아이콘 렌더링하는 경우에는 기획 혹은 디자인이 변경되었을 때 유연하게 대처가 어려울 것 같아요.  **구체적인 예를 들면, 새로운 기능이 추가되어 아이콘이 더 들어간다면 조건 분기가 더 많아질거에요.**

그래서 아이콘이 들어갈 위치만 해당 컴포넌트에서 정해주고 children으로 받는게 좋을 것 같다고 생각해요.

Reply:
@minjongbaek 

민종님 의견대로 default로 들어가게될 아이콘 "뒤로 가기", "관리자 드롭다운"만 남겨두고
이외의 아이콘들은 children으로 받는게 좋겠네요!!